src/Http/RequestFactory.php

```diff
@@ -194,30 +194,54 @@ public function createHttpRequest()
 		$usingTrustedProxy = $remoteAddr && array_filter($this->proxies, function ($proxy) use ($remoteAddr) {
 			return Helpers::ipMatch($remoteAddr, $proxy);
 		});
-
 		if ($usingTrustedProxy) {
-			if (!empty($_SERVER['HTTP_X_FORWARDED_PROTO'])) {
-				$url->setScheme(strcasecmp($_SERVER['HTTP_X_FORWARDED_PROTO'], 'https') === 0 ? 'https' : 'http');
-			}
+			if(!empty($_SERVER['HTTP_FORWARDED'])) {
+				$forwardParams = preg_split('/[,]|[;]/', $_SERVER['HTTP_FORWARDED']);
```

Comment:
Semicolin is used as a forward-pair delimiter whereas comma is used as a field separator. I'm afraid current implementation won't be able to distinguish it...